src/Fixer/ControlStructure/SingleExpressionPerLineFixer.php

```diff
@@ -0,0 +1,357 @@
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([new AllowedValueSubset([
+                    self::ELEMENTS_ARRAYS,
+                    self::ELEMENTS_ARGUMENTS,
+                    self::ELEMENTS_PARAMETERS,
+                    self::ELEMENTS_CONTROL_STRUCTURES,
+                    self::SWITCH_CASES,
+                    self::MATCH_EXPRESSIONS,
+                ])])
+                ->setDefault([
+                    self::ELEMENTS_ARRAYS,
+                    self::ELEMENTS_ARGUMENTS,
+                    self::ELEMENTS_PARAMETERS,
+                    self::ELEMENTS_CONTROL_STRUCTURES,
+                    self::SWITCH_CASES,
+                    self::MATCH_EXPRESSIONS,
+                ])
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $this->processBlock($tokens, 0, $tokens->count() - 1, false);
+    }
+
+    /**
+     * Process tokens from $tokens[$begin] to $tokens[$end] and add missing new lines:
+     * - If the code is not multiline, there is nothing to add
+     * - If the code is multiline, it looks for newline right after the first token,
+     * after every comma, and right before the last token and add them when they are missing.
+     *
+     * Since recursion is used to handle things like array inside array or functions we need
+     * a parameter $shouldAddNewLine to knows if we are currently fixing the structure, which
+     * requires to add newlines, or if we are just looking for the next one to fix.
+     *
+     * @param null|array<string> $lineDelimiters
+     * @param null|array<int>    $lineDelimitersIndexes
+     */
+    private function processBlock(
+        Tokens $tokens,
+        int $begin,
+        int $end,
+        bool $shouldAddNewLine,
+        ?array $lineDelimiters = null,
+        ?array $lineDelimitersIndexes = null
+    ): int {
+        $tokenAddedCount = 0;
+        if (!$tokens->isPartialCodeMultiline($begin, $end)) {
+            return $tokenAddedCount;
+        }
+
+        if ($shouldAddNewLine) {
+            $tokenAddedCount += $this->addNewLineAfterIfNecessary($tokens, $begin);
+        }
+
+        $fixArrays = \in_array(self::ELEMENTS_ARRAYS, $this->configuration['elements'], true);
+        $fixArguments = \in_array(self::ELEMENTS_ARGUMENTS, $this->configuration['elements'], true);
+        $fixParameters = \in_array(self::ELEMENTS_PARAMETERS, $this->configuration['elements'], true);
+        $fixControlStructures = \in_array(self::ELEMENTS_CONTROL_STRUCTURES, $this->configuration['elements'], true);
+        $fixSwitchCases = \in_array(self::SWITCH_CASES, $this->configuration['elements'], true);
+        $fixMatch = \in_array(self::MATCH_EXPRESSIONS, $this->configuration['elements'], true);
+
+        for ($index = $begin + 1; $index < $end + $tokenAddedCount; ++$index) {
+            /** @var Token $token */
```

Comment:
This should not be necessary as `Tokens` extends `\SplFixedArray<Token>` (`@extends` tag).